src/Microsoft.VisualStudio.Services.Agent/Blob/DedupManifestArtifactClientFactory.cs

```diff
@@ -166,8 +170,12 @@ private DedupManifestArtifactClientFactory()
                         tracer,
                         cancellationToken);
 
+                    Uri baseAddress = connection.GetClient<DedupStoreHttpClient>().BaseAddress;
```

Comment:
This function looks so similar to the above one. Are we missing domain and hashtype?